app/models/user.rb

```diff
@@ -68,31 +60,28 @@ def rating_record
     client.find_rating_info(participant_id).body.fetch('ratingRecord', {})
   end
 
-  # This is a helper method for pulling mhv_correlation_id
-  def mhv_correlation_id
-    mhv_correlation_ids.first
-  end
-
   def can_access_user_profile?
     loa1? || loa2? || loa3?
   end
 
   def can_access_mhv?
-    loa3? && mhv_correlation_ids.length == 1
+    loa3? && mhv_correlation_id
   end
 
   def can_access_evss?
     edipi.present? && ssn.present? && participant_id.present?
   end
 
+  delegate :edipi, to: :mvi
+  delegate :icn, to: :mvi
+  delegate :mhv_correlation_id, to: :mvi
+  delegate :participant_id, to: :mvi
+  delegate :va_profile, to: :mvi
+
   private
 
-  def mhv_correlation_ids
-    return @mhv_correlation_ids if @mhv_correlation_ids
-    ids = mvi&.dig(:mhv_ids)
-    ids = [] unless ids
-    @mhv_correlation_ids = ids.map { |mhv_id| mhv_id.split('^')&.first }.compact
-    @mhv_correlation_ids
+  def mvi
+    @mvi ||= Mvi.from_user(self)
```

Comment:
@kreek can this ever return NilClass? (It doesn't appear so looking at the code). But just wanted to double check, because otherwise  methods like `can_access_mhv?` would fail for `undefined method mhv_correlation_id on Nil::NilClass`